onnxruntime/core/providers/cuda/shared_inc/cuda_utils.h

```diff
@@ -48,10 +48,23 @@ struct TArray {
     ORT_ENFORCE(size <= capacity, "TArray size was set to ", size, ", exeeding the capacity limit of ", capacity);
   }
 
+  TArray(const std::vector<T>& vec) : size_(static_cast<int32_t>(vec.size())), data_() {
```

Comment:
>size_(static_cast<int32_t>(vec.size())), data_() [](start = 38, length = 48)

Call the ctor above to avoid duplicated code:
TArray(const std::vector<T>& vec) : TArray(static_cast<int32_t>(vec.size())) {
  memcpy(...
}

Reply:
I just updated the PR as you suggested. Can you review again? Thanks!